File: smkn8picket-client/src/models/Responses/StudentProgressNoteResponse.ts
export default interface StudentProgressNoteResponse {
  id: number;
  schoolYearId: number;
  schoolYearName?: string
  studentId: number;
  studentName?: string
  teacherId: number
  teacherName?: string
  note: string;
  createdAt: Date
  updatedAt?: Date
}

File: SharedModel/Responses/StudentProgressNoteResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SharedModel.Responses
{
    public record StudentProgressNoteResponse(
        int Id,
        int SchoolYearId,
        string? SchoolYearName,
        int StudentId,
        string? StudentName,
        int TeacherId,
        string? TeacherName,
        StudentProgressNoteType ProgressType,
        string Note,
        DateTime CreatedAt,
        DateTime? UpdatedAt
        );


}
